federation-js/src/composition/__tests__/composeAndValidate.test.ts

```diff
@@ -99,22 +105,43 @@ function permutateList<T>(inputArr: T[]) {
   return result;
 }
 
+// This assertion function should be used for the sake of convenient type refinement.
+// It should not be depended on for causing a test to fail. If an error is thrown
+// from here, its use should be reconsidered.
+export function assertCompositionSuccess(
+  compositionResult: CompositionResult,
+  message?: string,
+): asserts compositionResult is CompositionSuccess {
+  if (compositionHasErrors(compositionResult)) {
+    throw new Error(message || 'Unexpected test failure');
+  }
+}
+
+// This assertion function should be used for the sake of convenient type refinement.
+// It should not be depended on for causing a test to fail. If an error is thrown
+// from here, its use should be reconsidered.
+export function assertCompositionFailure(
+  compositionResult: CompositionResult,
+  message?: string,
+): asserts compositionResult is CompositionFailure {
+  if (!compositionHasErrors(compositionResult)) {
+    throw new Error(message || 'Unexpected test failure');
+  }
+}
+
 it('composes and validates all (24) permutations without error', () => {
   permutateList([
     inventoryService,
     reviewsService,
     accountsService,
     productsService,
   ]).map((config) => {
-    const { errors } = composeAndValidate(config);
+    const compositionResult = composeAndValidate(config);
 
-    if (errors.length) {
-      console.error(
-        `Errors found with composition [${config.map((item) => item.name)}]`,
-      );
-    }
-
-    expect(errors).toHaveLength(0);
+    assertCompositionSuccess(
```

Comment:
This violates the intention expressed in the comments above. This can be wrapped with an `expect().not.toThrow()`.